### ISSUE TITLE
NoSuchMethodError: rxjava2 and SubscriptionHelper.isCancelled() method not available

### ISSUE THREAD

**Issue by himanshumps** [resilience4j/resilience4j#400]
Hi,

The isCancelled method has been removed since rxjava version 2.2.4. We are getting NoSuchMethodError when using resilience4j-rxjava2-0.13.2.jar file. The stack trace is:

```
Caused by: java.lang.NoSuchMethodError: io/reactivex/internal/subscriptions/SubscriptionHelper.isCancelled(Lorg/reactivestreams/Subscription;)Z (loaded from jar:file:/product_standard_product_definition_api/target/app.jar!/BOOT-INF/lib/rxjava-2.2.6.jar!/ by org.springframework.boot.loader.LaunchedURLClassLoader@29294a18) called from class io.github.resilience4j.circuitbreaker.operator.CircuitBreakerSubscriber (loaded from jar:file:/product_standard_product_definition_api/target/app.jar!/BOOT-INF/lib/resilience4j-rxjava2-0.13.2.jar!/ by org.springframework.boot.loader.LaunchedURLClassLoader@29294a18).
        at io.github.resilience4j.circuitbreaker.operator.CircuitBreakerSubscriber.notCancelled(CircuitBreakerSubscriber.java:95)
        at io.github.resilience4j.circuitbreaker.operator.CircuitBreakerSubscriber.isInvocationPermitted(CircuitBreakerSubscriber.java:91)
        at io.github.resilience4j.circuitbreaker.operator.CircuitBreakerSubscriber.onNext(CircuitBreakerSubscriber.java:46)
        at io.reactivex.internal.util.HalfSerializer.onNext(HalfSerializer.java:45)
        at io.reactivex.internal.subscribers.StrictSubscriber.onNext(StrictSubscriber.java:97)
        at hu.akarnokd.rxjava.interop.ObservableV1ToFlowableV2$ObservableSubscriber.onNext(ObservableV1ToFlowableV2.java:62)
        at rx.internal.operators.OnSubscribeMap$MapSubscriber.onNext(OnSubscribeMap.java:77)
        at rx.internal.operators.OperatorMerge$MergeSubscriber.emitScalar(OperatorMerge.java:395)
        at rx.internal.operators.OperatorMerge$MergeSubscriber.tryEmit(OperatorMerge.java:355)
        at rx.internal.operators.OperatorMerge$InnerSubscriber.onNext(OperatorMerge.java:846)
        at rx.internal.operators.OnSubscribeSwitchIfEmpty$AlternateSubscriber.onNext(OnSubscribeSwitchIfEmpty.java:153)
        at rx.internal.operators.OnSubscribeMap$MapSubscriber.onNext(OnSubscribeMap.java:77)
        at rx.internal.operators.NotificationLite.accept(NotificationLite.java:135)
        at rx.internal.operators.OperatorOnBackpressureBuffer$BufferSubscriber.accept(OperatorOnBackpressureBuffer.java:156)
        at rx.internal.util.BackpressureDrainManager.drain(BackpressureDrainManager.java:198)
        at rx.internal.operators.OperatorOnBackpressureBuffer$BufferSubscriber.onNext(OperatorOnBackpressureBuffer.java:151)
        at rx.observers.SafeSubscriber.onNext(SafeSubscriber.java:134)
        at rx.internal.operators.NotificationLite.accept(NotificationLite.java:135)
        at rx.internal.operators.BufferUntilSubscriber$OnSubscribeAction.call(BufferUntilSubscriber.java:108)
        at rx.internal.operators.BufferUntilSubscriber$OnSubscribeAction.call(BufferUntilSubscriber.java:80)
        at rx.Observable.subscribe(Observable.java:10423)
        at rx.Observable.subscribe(Observable.java:10390)
        at com.couchbase.client.core.utils.UnicastAutoReleaseSubject$OnSubscribeAction.call(UnicastAutoReleaseSubject.java:228)
        at com.couchbase.client.core.utils.UnicastAutoReleaseSubject$OnSubscribeAction.call(UnicastAutoReleaseSubject.java:207)
        at rx.internal.operators.OnSubscribeLift.call(OnSubscribeLift.java:48)
        at rx.internal.operators.OnSubscribeLift.call(OnSubscribeLift.java:30)
        at rx.Observable.unsafeSubscribe(Observable.java:10327)
        at rx.internal.operators.OnSubscribeMap.call(OnSubscribeMap.java:48)
        at rx.internal.operators.OnSubscribeMap.call(OnSubscribeMap.java:33)
        at rx.Observable.unsafeSubscribe(Observable.java:10327)
        at rx.internal.operators.OnSubscribeSwitchIfEmpty$ParentSubscriber.subscribe(OnSubscribeSwitchIfEmpty.java:101)
        at rx.internal.operators.OnSubscribeSwitchIfEmpty.call(OnSubscribeSwitchIfEmpty.java:52)
        at rx.internal.operators.OnSubscribeSwitchIfEmpty.call(OnSubscribeSwitchIfEmpty.java:31)
        at rx.Observable.unsafeSubscribe(Observable.java:10327)
        at rx.internal.operators.OperatorMerge$MergeSubscriber.onNext(OperatorMerge.java:248)
        at rx.internal.operators.OperatorMerge$MergeSubscriber.onNext(OperatorMerge.java:148)
        at rx.internal.operators.OnSubscribeMap$MapSubscriber.onNext(OnSubscribeMap.java:77)
        at rx.internal.operators.OnSubscribeDoOnEach$DoOnEachSubscriber.onNext(OnSubscribeDoOnEach.java:101)
        at rx.internal.operators.OperatorMerge$MergeSubscriber.emitScalar(OperatorMerge.java:511)
        at rx.internal.operators.OperatorMerge$MergeSubscriber.tryEmit(OperatorMerge.java:466)
        at rx.internal.operators.OperatorMerge$MergeSubscriber.onNext(OperatorMerge.java:244)
        at rx.internal.operators.OperatorMerge$MergeSubscriber.onNext(OperatorMerge.java:148)
        at rx.internal.operators.OnSubscribeMap$MapSubscriber.onNext(OnSubscribeMap.java:77)
        at rx.observers.Subscribers$5.onNext(Subscribers.java:235)
        at rx.internal.operators.OnSubscribeDoOnEach$DoOnEachSubscriber.onNext(OnSubscribeDoOnEach.java:101)
        at rx.internal.operators.OperatorOnErrorResumeNextViaFunction$4.onNext(OperatorOnErrorResumeNextViaFunction.java:154)
        at rx.internal.operators.OnSubscribeTimeoutTimedWithFallback$TimeoutMainSubscriber.onNext(OnSubscribeTimeoutTimedWithFallback.java:120)
        at rx.observers.SafeSubscriber.onNext(SafeSubscriber.java:134)
        at rx.internal.producers.SingleProducer.request(SingleProducer.java:65)
        at rx.internal.producers.ProducerArbiter.setProducer(ProducerArbiter.java:126)
        at rx.internal.operators.OnSubscribeTimeoutTimedWithFallback$TimeoutMainSubscriber.setProducer(OnSubscribeTimeoutTimedWithFallback.java:155)
        at rx.Subscriber.setProducer(Subscriber.java:205)
        at rx.subjects.AsyncSubject.onCompleted(AsyncSubject.java:103)
        at com.couchbase.client.core.endpoint.AbstractGenericHandler.completeResponse(AbstractGenericHandler.java:508)
        at com.couchbase.client.core.endpoint.AbstractGenericHandler.access$000(AbstractGenericHandler.java:86)
        at com.couchbase.client.core.endpoint.AbstractGenericHandler$1.call(AbstractGenericHandler.java:526)
        at rx.internal.schedulers.ScheduledAction.run(ScheduledAction.java:55)
        ... 7 more
```

**Comment by RobWin** [comment]
Hi, 
Should be fixed in 0.140 which is released soon.
You can already try 0.14.0-SNAPSHOT.

**Comment by himanshumps** [comment]
We are using corporate nexus repository and it doesn't have snapshots enabled.

[WARNING] The POM for io.github.resilience4j:resilience4j-circuitbreaker:jar:0.14.0-SNAPSHOT is missing, no dependency information available
[WARNING] The POM for io.github.resilience4j:resilience4j-ratelimiter:jar:0.14.0-SNAPSHOT is missing, no dependency information available
[WARNING] The POM for io.github.resilience4j:resilience4j-retry:jar:0.14.0-SNAPSHOT is missing, no dependency information available
[WARNING] The POM for io.github.resilience4j:resilience4j-bulkhead:jar:0.14.0-SNAPSHOT is missing, no dependency information available
[WARNING] The POM for io.github.resilience4j:resilience4j-cache:jar:0.14.0-SNAPSHOT is missing, no dependency information available
[WARNING] The POM for io.github.resilience4j:resilience4j-timelimiter:jar:0.14.0-SNAPSHOT is missing, no dependency information available
[WARNING] The POM for io.github.resilience4j:resilience4j-rxjava2:jar:0.14.0-SNAPSHOT is missing, no dependency information available

**Comment by RobWin** [comment]
We are using JCenter
http://resilience4j.github.io/resilience4j/#_snapshot_2

**Comment by himanshumps** [comment]
When are you planning to release 0.14.0 ?

**Comment by RobWin** [comment]
I hope within the next days. There are some important PRs still in review.

**Comment by RobWin** [comment]
0.14.0 was released

**Comment by Romeh** [comment]
@himanshumps can u please test it again to see if we can close this ticket ?

**Comment by himanshumps** [comment]
I tested and it works with 0.14.0-SNAPSHOT

**Comment by himanshumps** [comment]
I will check with 0.14.0 as well and update on the ticket

**Comment by RobWin** [comment]
thx

**Comment by himanshumps** [comment]
Validated and looks like it has been fixed with 0.14.1.
